keps/sig-storage/20191125-bucket-provisioning.md

```diff
@@ -0,0 +1,428 @@
+4. A driver will then provision the object storage and return the connection information, to be written to the Bucket.
+
+
+
+
+#### Sharing Dynamically Created Buckets
+
+Once a [Bucket](#Bucket) is created, it is discoverable by other users in the cluster.  
+
+5. In order to access the Bucket, a user may create a [Bucket](#bucket) that specifies the Bucket by name.
+
+> Note: Per [Non-Goals](#non-goals), access management is not within the scope of this KEP.  ACLs, access policies, and credentialing should be handled out of band.
+
+### Brownfield
+
+“Brownfield” describes any case where the bucket already exists in the cloud service and is *not yet represented* by a [Bucket](#Bucket) instance.  In order to publish the cloud service bucket to the cluster:
+
+1. The bucket should already be provisioned in the cloud service.
+1. A Bucket is created to represent the provisioned bucket.  
+1. Once done, Bucket API instances may be created by users, specifying the desired Bucket.
+
+>  Note: Per [Non-Goals](#non-goals), access management is not within the scope of this KEP.  ACLs, access policies, and credentialing should be handled out of band.  Buckets may be configured to whitelist access for specific namespaces.
+
+---
+
+## Provisioner Secrets
+
+Per [Non-Goals](#non-goals), it is not within the scope of the proposal to abstract IAM operations.  Instead, provisioner and user credentials should be provided to automation by admins or users.  
+
+To allow for flexibility in authorizing provisioner operations, credentials may be provided to the provisioner in several ways.
+
+- **Per Provisioner:** the Secret is used for all provisioning operations, for the lifetime of the provisioner.  These Secrets should be injected directly into the provisioner's container via [common Kubernetes patterns](https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/).
+
+Credentials may also be specified at more granular levels in order to allow for context dependent keys.  E.g. When authorization differs between BucketClasses or between individual operations.  This may be facilitated by defining a set of string keys which the core automation will be aware of, so that Secrets may be referenced by BucketClasses.  For example:
+
+```yaml
+cosi.io/provisioner-secret-name:
+cosi.io/provisioner-secret-namespace:
+```
+
+- **Per BucketClass:** A secret may be made specific to a BucketClass.  This suits cases where authorization may be segregated in the object store.  The Secret may then be defined explicitly in the `bucketClass.parameters` map.
+
+  ```yaml
+  cosi.io/provisioner-secret-name: "foo"
+  cosi.io/provisioner-secret-namespace: "bar"
+  ```
+
+- **Per Operation/Bucket:** Unique credentials are passed per Bucket or operation. In order to support dynamic Secret naming, templating similar to [CSI Secret templating](https://kubernetes-csi.github.io/docs/secrets-and-credentials-storage-class.html) may be used.  E.g.
```

Comment:
If this is at all difficult or risky to implement I suggest saying that it is not a part of MVP.

Reply:
That is correct. This cannot be fulfilled as a part of the MVP